src/Estimators/SelfHealingOverlap.cpp

```diff
@@ -19,20 +20,54 @@
 namespace qmcplusplus
 {
 SelfHealingOverlap::SelfHealingOverlap(SelfHealingOverlapInput&& inp_, const TrialWaveFunction& wfn, DataLocality dl)
-    : OperatorEstBase(dl), input_(std::move(inp_))
+    : OperatorEstBase(dl), input_(std::move(inp_)), wf_type(no_wf)
 {
   //my_name_ = input_.get_name();
 
   auto& inp = this->input_.input_section_;
 
+  use_param_deriv = inp.get<bool>("param_deriv");
+
   auto msd_refvec = wfn.findMSD();
-  if (msd_refvec.size() != 1)
+  auto sd_refvec  = wfn.findSD();
+
+  auto nsd  = sd_refvec.size();
+  auto nmsd = msd_refvec.size();
+
+  size_t nparams;
+  if (nmsd==1 and nsd==0)
+  {// multi-slater-det wavefunction
+    wf_type = msd_wf;
+    const MultiSlaterDetTableMethod& msd = msd_refvec[0];
+    if (!use_param_deriv)
+      nparams = msd.getLinearExpansionCoefs().size();
+    else
+      nparams = msd.myVars->size();
+    if(nparams==0)
+      throw std::runtime_error("SelfHealingOverlap: multidet wavefunction has no parameters.");
+  }
+  else if (nmsd==0 and nsd==1)
+  {// slater-det wavefunction
+    throw std::runtime_error("SelfHealingOverlap: slaterdet wavefunction implementation incomplete");
+    //wf_type = sd_rot_wf;
+    //const SlaterDet& sd = sd_refvec[0];
+    //nparams = sd.myVars.size();
+    //if(nparams==0)
+    //  throw std::runtime_error("SelfHealingOverlap: slaterdet wavefunction has no parameters.\n  Please check that <rotated_sposet/>'s appear in the input file.");
```

Comment:
Please delete unused code. You can restore them from the history when needed.

Reply:
Done